design/src/rfcs/rfc0023_serialization_and_deserialization.md

```diff
@@ -0,0 +1,295 @@
+
+We believe that this will not be a problem.
+
+### Introduction of New Fields
+Most fields are `Option<T>` type. 
+When user de-serializes data written for a format before the new fields were introduced, new field will be assigned with `None` type.
+
+If a field isn't `Option`, `serde` uses `Default` trait to generate data to fill the field.  
+We believe that we could 
+
+If the new field is not an `Option<T>` type and has no `Default` implementation, we must implement a custom de-serialization logic.
+
+In case of serilization, introduction of new fields will not be an issue unless the data format requires a schema. (e.g. parquet, avro) However, this is outside of our responsibility.
+
+## Introduction of New Data Type
+If new field introduces a new data type, it will not require any additional work if the data type can derive `serde` traits.  
+
+If the data cannot derive `serde` traits on it's own, then we have two options.
+To clarify, this is the same approach we took on `Data Type to skip` section.
+1. skip
+   We will simply skip serializing/de-serializing. However, we may need to implement custom serialization/de-serialization logic if a value is not wrapped with `Option`.
+2. custom serialization/de-serialization logic 
+   We can implement tailored serialization/de-serialization logic, just like in 
+
+e.g.
+```rust
+#[derive(serde::Serialize, serde::Deserialize)]
+struct OutputV1 {
+  string_field: Option<String>
+}
+
+#[derive(serde::Serialize, serde::Deserialize)]
+struct OutputV2 {
+  string_field: Option<String>,
+  // this will always be treated as None value by serde
+  #[serde(skip)] 
+  skip_not_serializable: Option<SomeComplexDataType>,
+  // We can implement a custom serialization logic
+  #[serde(serialize_with = "custom_serilization_logic", deserialize_with = "custom_deserilization_logic")]
+  not_derive_able: SomeComplexDataType,
+  // Serialization will be skipped, and de-serialization will be handled with the function provided on default tag
+  #[serde(skip, default = "default_value")]
+  skip_with_custom: DataTypeWithoutDefaultTrait,
+}
+```
+# Discussions
+
+## Serialization and de-serialization support for an entire response/request
+The problem with serialization/de-serialization of an entire response/request the lack of data type that can be mapped to `serde`'s data model field by field.
+
+Currently, SDK has no data type that represents an entire response or request that can be mapped to `serde`'s data model; Thus, you must introduce a schema and implement logics that allows users to serialize/de-serialize their data.
+
+Although this RFC does not solve this issue, we believe that this RFC will help future contirbutor who wishes to implement serialization and de-serialization support for an entire response/request.
+
+
+## Sensitive Information
+If serialized data contains sensitive information, it will not be masked.  
+Users must be cautious to avoid compromisation
```

Comment:
There is a way. You can check the member shape or it's containing shape for the `SensitiveTrait`. The code would look similar to this:

```kt
/**
 * If the root shape is sensitive, then all member shapes are considered sensitive. Otherwise, iterate over member
 * shapes and filter out any missing the `SensitiveTrait`.
 */
private fun findShapeSensitivity(rootShape: Shape): List<MemberShape> = if (rootShape.hasTrait<SensitiveTrait>()) {
    rootShape.members().toList()
} else {
    rootShape
        .members()
        .filter { it.hasTrait<SensitiveTrait>() }
}
```

Reply:
That's cool. thanks!